provider/pkg/provider/provider.go

```diff
@@ -2364,19 +2366,21 @@ func (k *kubeProvider) Update(
 			// If it's a "no match" error, this is probably a CustomResource with no corresponding
 			// CustomResourceDefinition. This usually happens if the CRD was not created, and we
 			// print a more useful error message in this case.
-			return nil, pkgerrors.Wrapf(
-				awaitErr, "update of resource %q failed because the Kubernetes API server "+
+			return nil, fmt.Errorf(
+				"update of resource %q failed because the Kubernetes API server "+
 					"reported that the apiVersion for this resource does not exist. "+
-					"Verify that any required CRDs have been created", urn)
+					"Verify that any required CRDs have been created: %w", urn, awaitErr)
+
 		}
 
 		var getErr error
 		initialized, getErr = k.readLiveObject(oldLive)
 		if getErr != nil {
 			// Object update/creation failed.
-			return nil, pkgerrors.Wrapf(
-				awaitErr, "update of resource %q failed because the Kubernetes API server "+
-					"reported that it failed to fully initialize or become live", urn)
+			return nil, fmt.Errorf(
+				"update of resource %q failed because the Kubernetes API server "+
+					"reported that it failed to fully initialize or become live: %w", urn, getErr)
+
 		}
```

Comment:
This was returning `awaitErr` instead of `getErr`.

Reply:
While we're cleaning up errs, the original error message also seems like it could be improved upon as well. The code comment directly below: `but failed to initialize` conflicts with the error message we're returning: `reported that it failed to fully initialize or become live`. Something along the lines of object creation failed seems to be more fitting, since we send the patch update, but the object still does not exist on the cluster.